### PR TITLE
chore: replace JS for TS

### PR DESCRIPTION
**Pull request by moog16**
related #4225

I'm fairly certain I found all instances of `.js` that needed to be replaced to `.ts`. I ignored all demos/tests since we aren't rewriting those just yet.

TODO:
- revisit the `check-imports.js` script

### CODE REVIEW

File: docs/authoring-components.md [hunk 1]
```diff
@@ -738,7 +733,7 @@ class MDCNewComponentFoundation extends MDCFoundation {
 ```
 
 The primary purpose of this is so that our components can interoperate with aspects of Google's
-front-end infrastructure, such as Closure Stylesheets' [CSS Class Renaming](https://github.com/google/closure-stylesheets#renaming) mechanism. In addition, it
```
Comment (kfranqueiro) [inline docs/authoring-components.md]: This comment is not made irrelevant by the TypeScript conversion.

File: docs/code/architecture.md [hunk 2]
```diff
@@ -29,6 +29,10 @@ Foundation and Adapter. This lets us reuse Foundation code across multiple web
 platforms, e.g. React and Angular, by re-implementing only the Adapter. For now
 we've only implemented a vanilla JavaScript version of the Adapter.
 
+### TypeScript
+
+MDC Web has built our components using TypeScript to increase developer communication and spped. It's proven to catch bugs before releasing new versions of our components, and provides another layer of safety via types. TypeScript helps the Open Source Contributors and maintainers alike with adding features and fixing bugs.
```
Comment (kfranqueiro) [inline docs/code/architecture.md]: MDC Web components are built using ...

File: docs/contribute/code_review.md [hunk 3]
```diff
@@ -24,7 +24,8 @@ of having too many open pull requests.
 ### Check The Obvious
 
 *  Do tests pass?
-*  Does the demo work?
+*  Are there coresponding screenshot tests?
```
Comment (kfranqueiro) [inline docs/contribute/code_review.md]: corresponding

File: docs/contribute/code_review.md [hunk 4]
```diff
@@ -38,6 +39,7 @@ APIs:
 *  CSS classes
 *  DOM structure
 *  Utility JavaScript Functions
+*  TypeScript Declaration Files
```
Comment (kfranqueiro) [inline docs/contribute/code_review.md]: Why was this added to this list? Everything else in this list is manually engineered; aren't d.ts automatically generated?

File: docs/getting-started.md [hunk 5]
```diff
@@ -149,7 +149,7 @@ And open http://localhost:8080 in a browser. You should see a blue “Hello Worl
 Now that you have webpack configured to compile Sass into CSS, let's include the Sass files for the Material Design button. First install the Node dependency:
 
 ```
-npm install --save-dev @material/button
```
Comment (kfranqueiro) [inline docs/getting-started.md]: Don't remove `--save-dev`. I'm pretty sure I stopped someone from doing this before (or had them revert it).

File: docs/getting-started.md [hunk 6]
```diff
@@ -230,38 +230,69 @@ Now run `npm start` again and open http://localhost:8080. You should see a Mater
...
 - [babel-preset-es2015](https://www.npmjs.com/package/babel-preset-es2015): Preset for compiling es2015
 - [babel-plugin-transform-object-assign](http://npmjs.org/package/babel-plugin-transform-object-assign), for IE 11 support
 
+> NOTE: You can decide to leave out `ts-loader` and `typescript` if you do not want to use TypeScript. However we do recommend you using TypeScript as it provides all the features of type safety. You can also use other TypeScript loaders such as [Awesome TypeScript Loader](https://www.npmjs.com/package/awesome-typescript-loader). We chose ts-loader because it is what we use to build our components.
+
 You can install all of them by running this command:
 
 ```
-npm install --save-dev babel-core@6 babel-loader@7 babel-preset-es2015 babel-plugin-transform-object-assign
+npm install --save-dev babel-core@6 babel-loader@7 babel-preset-es2015 babel-plugin-transform-object-assign ts-loader@3.5.0
```
Comment (kfranqueiro) [inline docs/getting-started.md]: Should typescript also be installed locally? It's not in this list...

Also, babel-plugin-transform-object-assign should no longer be needed (we should test that - I'm pretty sure all instances of Object.assign should no longer exist on the feat/typescript branch)

File: docs/getting-started.md [hunk 7]
```diff
@@ -230,38 +230,69 @@ Now run `npm start` again and open http://localhost:8080. You should see a Mater
...
+
+We also recommend installing typescript globally to run `tsc` commands from your CLI:
+
+```
+npm install -g typescript # this allows you to run `tsc --init` in a subsequent step
+```
+
+### Step 4: TypeScript setup
+
+Now we need to setup TypeScript through your `tsconfig.json` file. The easiest way is to use the `TypeScript CLI` known as `tsc`. If you install it globally found in the [Step 3](#step-3-installing-javascript-packages), you can run the follow:
```
Comment (kfranqueiro) [inline docs/getting-started.md]: follow -> following

File: docs/getting-started.md [hunk 8]
```diff
@@ -230,38 +230,69 @@ Now run `npm start` again and open http://localhost:8080. You should see a Mater
...
+
+
```
Comment (kfranqueiro) [inline docs/getting-started.md]: Remove extra newline

File: docs/getting-started.md (hunk 5 above)
Comment (moog16) [reply docs/getting-started.md]: Why are we telling people to add their non-build step deps to devDeps?  `dependencies` are modules which are also required at runtime.

File: docs/getting-started.md (hunk 6 above)
Comment (moog16) [reply docs/getting-started.md]: Hmm...ya i suppose it should. Below I also recommend installing globally, but it should be included in their deps in package.json

File: docs/contribute/code_review.md (hunk 4 above)
Comment (moog16) [reply docs/contribute/code_review.md]: true - removing!

File: docs/authoring-components.md (hunk 1 above)
Comment (moog16) [reply docs/authoring-components.md]: ok I'll add this back

File: docs/closure-compiler.md [hunk 9]
```diff
@@ -1,179 +0,0 @@
-<!--docs:
```
[the remark this answers is not in the document]
Comment (moog16) [reply docs/closure-compiler.md]: Well if we want to keep this as a best practices for writing foundations/adapter/components with TypeScript that makes sense. But a lot of this seems to be covered in the `authoring-components.md`. What do you think?

File: docs/importing-js.md [hunk 10]
```diff
@@ -26,12 +26,20 @@ of your built assets, you will want to explicitly reference the package's `index
...
+
 Note that in this case, you must ensure your build toolchain is configured to process and transpile MDC Web's modules
 as well as your own. You will also need to include babel's
 [`transform-object-assign`](https://www.npmjs.com/package/babel-plugin-transform-object-assign) plugin for IE 11 support.
 
 See the [Getting Started guide](getting-started.md) for more details on setting up an environment.
 
+#### TypeScript
+
+If you are using TypeScript, MDC Web's packages also include `.d.ts` files for your consumption. Most of the time you shouldn't need to reference these, as the TypeScript compiler should automatically pick them up. MDC Web has also defined the `types` property found in `package.json`, for your convinence. There is a bundled `.d.ts` file found under the `dist` directory that maps to the respective UMD module. There are corresponding `.d.ts` files for each foundation/component/adapter/etc. within the package.
```
Comment (acdvorak) [inline docs/importing-js.md]: Spelling: `convinence`. Also, what does "for your convenience" mean? Why is the `types` property important; what does it do?

File: docs/importing-js.md [hunk 11]
```diff
@@ -26,12 +26,20 @@ of your built assets, you will want to explicitly reference the package's `index
...
+
+> NOTE: Intentionally we did not include the `.ts` source files in our packages. Everyone can use the `.d.ts` files and transpiled `.js` (in ES5 or ES2015+ format) as `.js` files are universally accepted.
```
Comment (acdvorak) [inline docs/importing-js.md]: Reword first sentence and add an explanation:

```markdown
NOTE: We intentionally omit `.ts` source files in our packages because <REASON>. ...
```

File: docs/contribute/code_review.md [hunk 12]
```diff
@@ -24,7 +24,8 @@ of having too many open pull requests.
 ### Check The Obvious
 
 *  Do tests pass?
-*  Does the demo work?
+*  Are there corresponding screenshot tests?
```
Comment (acdvorak) [inline docs/contribute/code_review.md]: Link to `test/screenshot/` directory

File: docs/code/architecture.md [hunk 13]
```diff
@@ -29,6 +29,10 @@ Foundation and Adapter. This lets us reuse Foundation code across multiple web
 platforms, e.g. React and Angular, by re-implementing only the Adapter. For now
 we've only implemented a vanilla JavaScript version of the Adapter.
 
+### TypeScript
+
+MDC Web components are built using TypeScript to increase developer communication and spped. It's proven to catch bugs before releasing new versions of our components, and provides another layer of safety via types. TypeScript helps the Open Source Contributors and maintainers alike with adding features and fixing bugs.
```
Comment (acdvorak) [inline docs/code/architecture.md]:
```markdown
MDC Web components are written in [TypeScript](https://www.typescriptlang.org/)
to increase developer velocity and reduce errors. Our npm releases include
ES5 and ES2015+ JavaScript bundles, as well as `.d.ts` typing declaration files
for TypeScript users. See [Importing JS](../importing-js.md) for more information.
```

File: docs/authoring-components.md [hunk 14]
```diff
@@ -760,16 +755,11 @@ Concretely:
...
-
-#### Closure Compatibility
 
-> NOTE: Our currently existing components are in the process of being made compatible with closure.
+#### TypeScript Compatibility
 
-All core MDC Web components must be fully compatible with the Google Closure Compiler using its
-advanced compilation mechanisms. We've provided a thorough explanation of this, as well as
-conventions, examples, and common closure patterns you may not be used to, in our [closure compiler documentation](./closure-compiler.md).
+All core MDC Web components must be fully compatible with the [TypeScript](https://www.typescriptlang.org/index.html). We've provided a list of requirements and further explanation of meeting the TypeScript compatibility, as well as
```
Comment (acdvorak) [inline docs/authoring-components.md]:
```markdown
All core MDC Web components must be fully compatible with strict-mode [TypeScript](https://www.typescriptlang.org/).
```

File: docs/authoring-components.md [hunk 15]
```diff
@@ -738,7 +733,7 @@ class MDCNewComponentFoundation extends MDCFoundation {
 ```
 
 The primary purpose of this is so that our components can interoperate with aspects of Google's
-front-end infrastructure, such as Closure Stylesheets' [CSS Class Renaming](https://github.com/google/closure-stylesheets#renaming) mechanism. In addition, it
+front-end infrastructure, such as Closure Stylesheets' [CSS Class Renaming](https://github.com/google/closure-stylesheets#renaming) mechanism.. In addition, it
```
Comment (acdvorak) [inline docs/authoring-components.md]: Remove duplicate `.`

File: docs/authoring-components.md [hunk 16]
```diff
@@ -478,9 +472,10 @@ A typical component within our codebase looks like so:
 packages
   ├── mdc-component
       ├── README.md # The component's README
-      ├── constants.js # The component's cssClasses/strings/numbers constants
-      ├── foundation.js # The component's foundation class
-      ├── index.js # The file that contains the vanilla component class, as well as exports the foundation
+      ├── adapter.ts # The adapter interface implemented by the component
+      ├── constants.ts # The component's cssClasses/strings/numbers constants
+      ├── foundation.ts # The component's foundation class
+      ├── index.ts # The file that contains the vanilla component class, as well as exports the foundation
```
Comment (acdvorak) [inline docs/authoring-components.md]: Add `component.ts` and update this comment to reflect that `index.ts` only re-exports the other files.

File: docs/authoring-components.md [hunk 17]
```diff
@@ -250,39 +246,35 @@ be used to build out lint tools to enforce proper adapter shape. This example sh
...
   static get defaultAdapter() {
     return {
       getAttr: (/* attr: string */) => /* string */ '',
-      setAttr: (/* attr: string, value: string */) => {},
-      addClass: (/* className: string */) => {},
-      removeClass: (/* className: string */) => {},
-      setToggleColorTextContent: (/* textContent: string */) => {},
-      registerInteractionHandler: (/* type: string, handler: EventListener */) => {},
-      deregisterInteractionHandler: (/* type: string, handler: EventListener */) => {}
+      setAttr: (/* attr: string, value: string */) => undefined,
```
Comment (acdvorak) [inline docs/authoring-components.md]: Replace Closure Doc comments with TypeScript

File: docs/authoring-components.md [hunk 18]
```diff
@@ -231,14 +232,9 @@ methods. We can now take these fake methods and transform them into an adapter i
...
-| SOMEHOW_REGISTER_INTERACTION_HANDLER(type: string, handler: EventListener) | registerInteractionHandler(type: string, handler: EventListener) |
-| SOMEHOW_DEREGISTER_INTERACTION_HANDLER(type: string, handler: EventListener) | deregisterInteractionHandler(type: string, handler: EventListener) |
 
-> Note: It is a convention in our code base to use the terms `registerInteractionHandler` and
-`deregisterInteractionHandler` as adapter methods for adding/removing generic event listeners. We do
-this because we feel it more semantic in that most components are only interested in interactivity.
-However, feel free to call these methods `{add,remove}EventListener` or anything else you would
-like.
+> Note: We have changed our convention of using `registerInteractionHandler` and
+`deregisterInteractionHandler` as adapter methods for adding/removing generic event listeners. Instead please manage events in the component layer. This is because event systems can change from framework/environment (ie. React Synthetic Events).
```
Comment (acdvorak) [inline docs/authoring-components.md]: Change this:

```markdown
Instead please manage events in the component layer.
This is because event systems can change from framework/environment
```

to this:

```markdown
Event handling varies significantly across frameworks
(e.g., [React Synthetic Events](https://reactjs.org/docs/events.html)),
so we recommend managing events in the component layer.
```

File: docs/authoring-components.md [hunk 19]
```diff
@@ -231,14 +232,9 @@ methods. We can now take these fake methods and transform them into an adapter i
...
 | SOMEHOW_UPDATE_TOGGLE_COLOR_TEXT_CONTENT(textContent: string) | setToggleColorTextContent(textContent: string) |
-| SOMEHOW_REGISTER_INTERACTION_HANDLER(type: string, handler: EventListener) | registerInteractionHandler(type: string, handler: EventListener) |
-| SOMEHOW_DEREGISTER_INTERACTION_HANDLER(type: string, handler: EventListener) | deregisterInteractionHandler(type: string, handler: EventListener) |
 
-> Note: It is a convention in our code base to use the terms `registerInteractionHandler` and
-`deregisterInteractionHandler` as adapter methods for adding/removing generic event listeners. We do
-this because we feel it more semantic in that most components are only interested in interactivity.
-However, feel free to call these methods `{add,remove}EventListener` or anything else you would
-like.
+> Note: We have changed our convention of using `registerInteractionHandler` and
```
Comment (acdvorak) [inline docs/authoring-components.md]: It would be clearer to phrase this more strongly:

```markdown
> Note: We no longer recommend using `registerInteractionHandler` and
```

File: docs/authoring-components.md (hunk 18 above)
Comment (acdvorak) [inline docs/authoring-components.md]: Nit: Change "ie." to "e.g.," ([Grammar Girl](https://www.quickanddirtytips.com/education/grammar/ie-versus-eg) to the rescue!)

File: docs/importing-js.md (hunk 10 above)
Comment (moog16) [reply docs/importing-js.md]: I read this doc again: https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

It reads, "Also note that if your main declaration file is named index.d.ts and lives at the root of the package (next to index.js) you do not need to mark the "types" property, though it is advisable to do so."

I've been reading this incorrectly, and its actually required in our case. I'll update the wording.

File: docs/authoring-components.md (hunk 17 above)
Comment (moog16) [reply docs/authoring-components.md]: I replaced it with:
```
      getAttr: (attr: string) => '',
      setAttr: (attr: string, value: string) => undefined,
      addClass: (className: string) => undefined,
      removeClass: (className: string) => undefined,
      setToggleColorTextContent: (textContent: string) => undefined,
      registerInteractionHandler: (type: string, handler: EventListener) => undefined,
      deregisterInteractionHandler: (type: string, handler: EventListener) => undefined
```

Although in our actual implementation we don't have unused arguments. Should I just remove them to look like:

```
      getAttr: () => '',
      setAttr: () => undefined,
      addClass: () => undefined,
      removeClass: () => undefined,
      setToggleColorTextContent: () => undefined,
      registerInteractionHandler: () => undefined,
      deregisterInteractionHandler: () => undefined,
```

**Comment by acdvorak** [comment]
We can probably update `check-imports.js` in PR #4449.

File: docs/authoring-components.md (hunk 17 above)
Comment (acdvorak) [reply docs/authoring-components.md]: I would probably leave them to make it clear that they exist, even if we're not using them.

File: docs/importing-js.md [hunk 20]
```diff
@@ -26,12 +26,20 @@ of your built assets, you will want to explicitly reference the package's `index
 import {MDCFoo, MDCFooFoundation} from '@material/foo/index';
 ```
 
+Certain build tools will often pick up on the use of `package.json`'s `module` property, which points to the ES2015+ `index.js` source code. If you are using [Webpack](https://webpack.js.org/) or [Rollup](https://rollupjs.org/guide/en), then you will not need to reference the `index.js` file, and can continue to use the `@material/foo` import paths.
```
Comment (acdvorak) [inline docs/importing-js.md]:
```markdown
Certain build tools will detect the `module` property in `package.json`,
which points to ES2015+ source code in an `index.js` file.
If you're using [Webpack](https://webpack.js.org/) or
[Rollup](https://rollupjs.org/guide/en), you do not need to
reference `/index` directly, and can continue to use the shorter
`@material/foo` import path syntax.
```

(Reformat the above into one line)

File: docs/authoring-components.md [hunk 21]
```diff
@@ -316,8 +310,16 @@ Since this component is a vanilla component, it should be modeled after the vani
 favors getters and setters to implement its functionality (think `checked`, `disabled`, etc.). Our
 adapter is extremely straightforward as we can simply repurpose the methods we started out with.
 
-```js
+```ts
 class RedblueToggle extends MDCComponent {
+  initialize() {
+    this.root.addEventListener('click', this.foundation_.handleClick);
```
Comment (kfranqueiro) [inline docs/authoring-components.md]: These can use `this.listen` / `this.unlisten` here now that the class extends MDCComponent

File: docs/authoring-components.md [hunk 22]
```diff
@@ -478,9 +474,11 @@ A typical component within our codebase looks like so:
 packages
   ├── mdc-component
       ├── README.md # The component's README
-      ├── constants.js # The component's cssClasses/strings/numbers constants
-      ├── foundation.js # The component's foundation class
-      ├── index.js # The file that contains the vanilla component class, as well as exports the foundation
+      ├── adapter.ts # The adapter interface implemented by the component
```
[the remark this answers is not in the document]
Comment (kfranqueiro) [reply docs/authoring-components.md]: Were some of Andy's comments not resolved?

File: docs/closure-compiler.md [hunk 23]
```diff
@@ -1,179 +0,0 @@
...
-  parameterized by `MDCRippleFoundation`, which - as shown above - is itself parameterized by
-  `MDCRippleAdapter`. Thus, the `MDCRipple` class is declared as
-  `class MDCRipple extends MDCComponent<MDCRippleFoundation>`.
-
-## MDC Web Closure Conventions
-
-The following guidelines outline the general conventions for writing closure-friendly code for MDC Web.
-This section should contain most - if not all - of what you need to get up and running writing components in our codebase.
-
-### All `import` statements must _not_ use re-exported modules.
```
Comment (kfranqueiro) [inline docs/closure-compiler.md]: This should effectively be moved to best_practices.md in the form of a heading explaining that in the context of component packages, only the index and component files are allowed to reference `index`, because wrappers shouldn't need to pull in `index` or `component`

File: docs/closure-compiler.md [hunk 24]
```diff
@@ -1,179 +0,0 @@
...
-interface EventDataType {
-  foo: string;
-  bar: number;
-}
-```
-<!--{% endraw %} -->
-
-Using this convention allows us to write tooling around handling these expressions, such as lint rule exceptions.
-
-### For key/value maps, use index signatures where possible
```
Comment (kfranqueiro) [inline docs/closure-compiler.md]: This should be moved to best_practices.md; I added this specifically for TypeScript

File: docs/importing-js.md (hunk 20 above)
Comment (kfranqueiro) [reply docs/importing-js.md]: It looks like this comment hasn't been applied.

Furthermore, index.js isn't ES2015+ source anymore. It's ES5 source in an ES module.

Comment (acdvorak) [reply docs/importing-js.md]: How would you phrase "ES5 source in an ES module" in context?

File: docs/authoring-components.md (hunk 22 above)
[the remark this answers is not in the document]
Comment (moog16) [reply docs/authoring-components.md]: I'm just about to fix these. I didn't see this yesterday.

File: docs/importing-js.md (hunk 20 above)
Comment (moog16) [reply docs/importing-js.md]: The 2 index.js files of the npm package are:

```js
dist/index.js (ES5) -- UMD
index.js (ES2015+) -- ESModule
```

Isn't what Andy wrote correct?

Comment (kfranqueiro) [reply docs/importing-js.md]: The distinction is that prior to the TS conversion, *.js was ES modules _with ES2015+ syntax_, thus requiring babel and additionally the object assign polyfill. After the TS conversion, *.js are ES modules _with ES5 syntax_. I believe that means you can get away with just webpack and not babel at all?

In terms of phrasing, something like: "which points to an ES module otherwise containing only ES5 syntax". We could link something to Rollup's pkg.module definition at https://github.com/rollup/rollup/wiki/pkg.module if we want.

File: docs/closure-compiler.md (hunk 23 above)
Comment (moog16) [reply docs/closure-compiler.md]: Is this line still accurate? 

This is an unfortunate side-effect of how [closure's module naming mechanism works](https://github.com/google/closure-compiler/issues/2257).

File: docs/closure-compiler.md (hunk 24 above)
Comment (moog16) [reply docs/closure-compiler.md]: Re both closure-compiler comments: I ended up adding this entire block:
```

#### All `import` statements must _not_ use re-exported modules

Only the `index.ts` or `component.ts` files are allowed to reference from other component packages' `index.ts`. This is because wrapping libraries only use `foundation` and `adapter`, so we should decouple the `component`.

```ts
// BAD
import {MDCFoundation} from '@material/base';
// GOOD
import {MDCFoundation} from '@material/base/foundation';
```

#### All adapters must be defined as interfaces

Each adapter must be defined within an `adapter.ts` file in the component's package directory.
All methods should contain a summary of what they should do. This summary should be
copied over to the adapter API documentation in our README. This will facilitate future endeavors
to potentially automate the generation of our adapter API docs. _Note that this replaces the
inline comments present in the methods within `defaultAdapter`_.

 ```ts
// adapter.ts
export interface MDCComponentAdapter {
  /**
   * Adds a class to the root element.
   */
  addClass(className: string): void;
   /**
   * Removes a class from the root element.
   */
  removeClass(className: string): void;
}
```

#### Foundation classes must extend `MDCFoundation`

 Foundations must extend `MDCFoundation` parameterized by their respective adapter. The
`defaultAdapter` must return an object with the correct adapter shape.

```ts
// foundation.ts
import {MDCFoundation} from '@material/base/foundation';
import MDCComponentAdapter from './adapter';
export class MDCComponentFoundation extends MDCFoundation<MDCComponentAdapter> {
  static get defaultAdapter(): MDCComponentAdapter {
    return {
      addClass: (className: string) => undefined,
      removeClass: (className: string) => undefined,
    };
  }
}
```

#### Component classes must extend `MDCComponent`

Components must extend `MDCComponent` parameterized by their respective foundation.

```ts
// index.ts
import {MDCComponent} from '@material/base/component';
import MDCComponentFoundation from './foundation';
export class MDCAwesomeComponent extends MDCComponent<MDCComponentFoundation> {
  getDefaultFoundation(): MDCComponentFoundation {
    return new MDCComponentFoundation({
      addClass: (className: string) => this.root_.classList.add(className),
      removeClass: (className: string) => this.root_.classList.remove(className),
    });
  }
}
```

### For key/value maps, use index signatures where possible

Index signatures are useful for homogeneous maps of key/value pairs, while interfaces are useful for specific object signatures.
Both are more specific than just typing something as `object`, and are thus preferable for their respective use cases.

```ts
static get strings(): {[key: string]: string} {
  ...
}
```

File: docs/importing-js.md (hunk 20 above)
Comment (moog16) [reply docs/importing-js.md]: @kfranqueiro I just checked by running `npm run build:esmodules`, so I do think you're correct.

Also I just noticed that we don't use `Object.assign` anywhere in our `packages/` directory. I do not think you need babel for either of our UMD or ESM packages.

Comment (kfranqueiro) [reply docs/importing-js.md]: RE your last point, that's what I mean. Previously we were relying on the object assign polyfill to generate our UMD (you were never expected to need babel to consume those). Now we don't need to use that polyfill for our build process, and neither does anyone else using us within their own.

Comment (moog16) [reply docs/importing-js.md]: In getting-started.md there is this dep:
```
- [babel-plugin-transform-object-assign](http://npmjs.org/package/babel-plugin-transform-object-assign), for IE 11 support
```

I am going to remove it
